api/api-facade/api-http/src/main/java/org/eclipse/dirigible/api/v3/http/HttpClientFacade.java

```diff
@@ -206,13 +207,20 @@ public static HttpClientResponse processHttpClientResponse(CloseableHttpResponse
 			if (entity != null && entity.getContent() != null) {
 				byte[] content = IOUtils.toByteArray(entity.getContent());
 
-				if ((ContentType.getOrDefault(entity).getMimeType().equals(ContentType.TEXT_PLAIN.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.TEXT_HTML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.TEXT_XML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_JSON.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_ATOM_XML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_XML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_XHTML_XML.getMimeType())) && (!binary)) {
+				 HashMap<String, Boolean> textMimeTypes = new HashMap<String, Boolean>();
```

Comment:
I wrongly commented this should be a `HashMap`, I meant `HashSet`, sorry for the confusion. The other part of the comment still stands - this should be a constant instead of created every time `processHttpClientResponse` is invoked.

Reply:
Thank you for the fix. Pushed to PR.